src/sonic_ax_impl/mibs/ietf/rfc2737.py

```diff
@@ -190,14 +192,14 @@ def _update_per_namespace_data(self, statedb, pubsub):
 
         # This code is not executed in unit test, since mockredis
         # does not support pubsub
-        if not pubsub:
-            redis_client = statedb.get_redis_client(statedb.STATE_DB)
-            db = statedb.get_dbid(statedb.STATE_DB)
-            pubsub = redis_client.pubsub()
-            pubsub.psubscribe("__keyspace@{}__:{}".format(db, self.TRANSCEIVER_KEY_PATTERN))
+        if not self.pubsub[inst]:
+            redis_client = self.statedb[inst].get_redis_client(self.statedb[inst].STATE_DB)
+            db = self.statedb[inst].get_dbid(self.statedb[inst].STATE_DB)
+            self.pubsub[inst] = redis_client.pubsub()
+            self.pubsub[inst].psubscribe("__keyspace@{}__:{}".format(db, self.TRANSCEIVER_KEY_PATTERN))
```

Comment:
Duplicated code. Could you reuse?

Reply:
This similar pubsub update is done only 2 instances with different DB Name and pattern.

Reply:
Duplicated with ieee802_1ab.py

---
In reply to: [441920582](https://github.com/Azure/sonic-snmpagent/pull/134#discussion_r441920582) [](ancestors = 441920582)

Reply:
Fixed as per review comments.